gnocchi/storage/_carbonara.py

```diff
@@ -396,8 +403,10 @@ def _compute_and_store_timeseries(self, metric, measures):
 
         agg_methods = list(metric.archive_policy.aggregation_methods)
         block_size = metric.archive_policy.max_block_size
-        back_window = metric.archive_policy.back_window
         definition = metric.archive_policy.definition
+        # NOTE(sileht): We keep one more blocks to calculate rate of change
+        # correctly
+        back_window = metric.archive_policy.back_window + 1
```

Comment:
:( can we do this only if rate is being computed? i don't think it makes sense to double the back_window size unnecessarily.